packages/@aws-cdk/aws-codestarnotifications/lib/notification-rule.ts

```diff
@@ -0,0 +1,189 @@
+import { IResource, Resource, Names } from '@aws-cdk/core';
+import * as constructs from 'constructs';
+import { CfnNotificationRule } from './codestarnotifications.generated';
+import { INotificationRuleSource } from './notification-rule-source';
+import { INotificationRuleTarget, NotificationRuleTargetConfig } from './notification-rule-target';
+
+/**
+ * The level of detail to include in the notifications for this resource.
+ */
+export enum DetailType {
+  /**
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch
+   */
+  BASIC = 'BASIC',
+
+  /**
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   */
+  FULL = 'FULL',
+}
+
+/**
+ * Standard set of options for `notifyOnXxx` codestar notification handler on construct
+ */
+export interface NotificationRuleOptions {
+  /**
+   * The name for the notification rule.
+   * Notification rule names must be unique in your AWS account.
+   *
+   * @default - generated from the `id`
+   */
+  readonly notificationRuleName?: string;
+
+  /**
+   * The status of the notification rule.
+   * If the enabled is set to DISABLED, notifications aren't sent for the notification rule.
+   *
+   * @default true
+   */
+  readonly enabled?: boolean;
+
+  /**
+   * The level of detail to include in the notifications for this resource.
+   * BASIC will include only the contents of the event as it would appear in AWS CloudWatch.
+   * FULL will include any supplemental information provided by AWS CodeStar Notifications and/or the service for the resource for which the notification is created.
+   *
+   * @default DetailType.FULL
+   */
+  readonly detailType?: DetailType;
+}
+
+/**
+ * Properties for a new notification rule
+ */
+export interface NotificationRuleProps extends NotificationRuleOptions {
+  /**
+   * A list of event types associated with this notification rule.
+   * For a complete list of event types and IDs, see Notification concepts in the Developer Tools Console User Guide.
+   * @see https://docs.aws.amazon.com/dtconsole/latest/userguide/concepts.html#concepts-api
+   *
+   * @default - No events.
+   */
+  readonly events?: string[];
```

Comment:
This should be required, no? Is it even allowed to create a Rule with no events? So basically a Rule that will never fire?

Reply:
Yes, It should be required, I will adjust!